test/demo/utils/common.ts

```diff
@@ -0,0 +1,110 @@
+import {Identity} from 'openfin/_v2/main';
+
+import {Intent} from '../../../src/client/intents';
+import {testManagerIdentity} from '../constants';
+
+import {fin} from './fin';
+import * as fdc3Remote from './fdc3RemoteExecution';
+import {delay} from './delay';
+
+export type Boxed<T> = {value: T}
+
+/**
+ * Identity with a mandatory `name`
+ */
+export interface AppIdentity extends Identity {
+    name: string; // Make inherited field mandatory
+}
+
+export interface DirectoryAppIdentity extends AppIdentity {
+    appId: string;
+}
+
+export interface NonDirectoryAppIdentity extends AppIdentity {
+    manifestUrl: string;
+}
+
+/**
+ * Quits an OpenFin app (or multiple in parallel) using `force=true` and swallowing errors
+ * @param apps Identities of the apps to quit
+ */
+export async function quitApp(...apps: Identity[]) {
+    await Promise.all(apps.map(app => fin.Application.wrapSync(app).quit(true).catch(() => {})));
+}
+
+export async function waitForAppToBeRunning(app: Identity): Promise<void> {
+    while (!await fin.Application.wrapSync(app).isRunning()) {
+        await delay(100);
+    }
+
+    // Additional delay to ensure app window is ready for puppeteer connection
+    await delay(500);
```

Comment:
I wouldn't expect this delay to be necessary, but if we see problems without it then I guess its fine to leave it.

Reply:
Yeah, seeing problems without this. Before, the timeout in the loop was 500, which meant that *most* of the time, the app was given long enough

Reply:
Hmmm, fair enough. Its a one-off so not a huge problem.